docs/design-proposals/multi-tenancy/multi-tenancy-overview.md

```diff
@@ -0,0 +1,149 @@
+# Arktos Multi-Tenancy Overview
+
+Xiaoning Ding, Qian Chen, Peng Du, Hong Chen
+
+## Introduction
+
+This document gives an overview of the multi-tenancy design in Arktos, including its motivation, requirements, design overview and lists of components changes and execution plans. For the detailed design of each part, please refer to the other documents in the same document directory.
+
+## Motivation
+
+Kubernetes provides a solution for automating deployment, scaling, and operations of application containers across clusters of hosts. However, it does not have support multi-tenancy. There is no explicit “tenant” concept in the system, and there is no resource isolation, usage isolation or performance isolation for different tenants.
+
+Arktos is evolved from Kubernetes. The motivation of Arktos multi-tenancy is to enable multiple organizations or teams to safely share a physical Arktos cluster, where there could be no trust among these organizations or teams. 
+
+The targeted scenarios include:
+
+* Provide an Arktos service to public cloud users.
+* Share a same Arktos cluster in private cloud, among organizations or teams that require strong isolation.
+
+## Requirements
+
+There are several different multi-tenancy models. Kubernetes community has [some documents](https://docs.google.com/document/d/15w1_fesSUZHv-vwjiYa9vN_uyc--PySRoLKTuDhimjc/edit#heading=h.3dawx97e3hz6) to compare these models. In general, they can be categorized as “soft” models or “hard” models. Soft models are for scenarios of private cloud or within an organization, where there is a certain level of trust between different tenants. While hard model is for situations where there is no trust among tenants and hard model provides strong isolation.
+
+Since Arktos aims to support the scenarios of no tenant trust, Arktos multi-tenancy is a hard multi-tenancy model. Meanwhile, flexible cross-tenant access may be also needed. A typical scenario is that team A allows team B to access its certain resources, but not all other teams.
+
+The following functionality requirements are derived from the above discussion:
+
+* **Isolation**: tenants should be strongly isolated from each other. They are not aware of the existence of others at all. One tenant cannot access another tenant's resources or impact other tenants' performance. 
```

Comment:
The description of "Isolation" conflicts with the "Flexible cross-tenant access" concept above. 

I remember in our discussion we put strong complete isolation as PH-1 goal, while flexible cross-tenant access as PH-2 target. So I think we can make it clear here.

Reply:
Sure, I will make the wording more clear here.

Actually they don't conflict with each other.  **By default** tenants are isolated from each other, but a tenant admin can set policies allowing other tenants to access its API resources.  

This functionality requirement covers both phase-1 or phase-2.

Reply:
Updated the wording.